### PR TITLE
feat(0158): delegate dry-run run-as validation to risktypes

### PR DESCRIPTION
**Pull request by isseis**
## Summary

- Delegate dry-run's user/group identity validation to `risktypes.ResolveRunAsIdentStrict`, the same fail-closed judgment the executor applies at real execution time, so a dry-run "validated" report can never diverge from what execution would actually do.
- Remove the dependency on `privilegeManager.WithPrivileges(OperationUserGroupDryRun)` for validation: validation now runs unconditionally, regardless of privilege-manager availability (previously it was skipped entirely when unsupported).
- Raise `Impact.SecurityRisk` monotonically on validation failure (`max` of current and `high`), fixing a pre-existing defect where a `critical` verdict could be overwritten down to `high`.
- Emit one structured `slog` record per validated command (`dry_run`, `command`, `group`, `run_as_user`, `run_as_group`, plus `resolved_uid`/`resolved_gid` on success or `failure_kind`/`error` on failure), with no environment-variable or argument values leaked into the log.
- Add AST-based "identity mutation guard" tests (empty allow-list) to the `resource` and `risktypes` packages, sharing scan logic via a new `internal/testutil/identitymutationguard` package.

## Review points

- Does the monotonic risk raise (`max`-based, never lowering) match the design (§3.4)?
- Does `runAsFailureKind`'s classification order use only `errors.Is`/`errors.AsType` (never a direct `==` comparison)?
- Is the supplementary-group-enumeration-failure regression (previously reported as "validated", now correctly reported as a failure) intentional and covered by a test?
- Do the structured log attributes exclude environment-variable values and command arguments (secret-leak prevention)?
- Do the 6 rewritten existing subtests and 10 new tests each check a distinct, intended invariant (no redundant coverage, no gaps)?

This PR went through one internal review pass (a Major finding — duplicated AST-scan helpers across the two guard test files — was fixed by extracting shared logic into `internal/testutil/identitymutationguard`; a Minor finding — missing negative assertions in two subtests — was also fixed) plus a verification pass confirming both fixes and no regressions.

### CODE REVIEW

File: internal/testutil/identitymutationguard/helpers.go [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+	entries, err := os.ReadDir(dir)
+	require.NoErrorf(t, err, "failed to read directory %s", dir)
+
+	var sites []CallSite
+	var valueRefs []ValueRef
+	for _, entry := range entries {
+		name := entry.Name()
+		if entry.IsDir() || !strings.HasSuffix(name, ".go") || strings.HasSuffix(name, "_test.go") {
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/testutil/identitymutationguard/helpers.go:105]: The referenced identity_mutation_guard_test.go does the opposite of what this claims: its doc comment (lines 298-313) explicitly says a constraint merely satisfiable with the tag, like "test || performance", still builds under "performance" alone and is therefore deliberately NOT exempted -- isTestHelpersFileName is only a one-directional assertion that named helper files carry a test-only constraint, never a basis for skipping. helpers.go's FindRefs mirrors that same intentional design (identical isTestOnlyBuildConstrained/constraintRequiresTag logic), so it already aligns with the guard rather than diverging from it.
